#VRML_SIM R2025a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/floors/protos/RectangleArena.proto"

WorldInfo {
}
Viewpoint {
  orientation 0.2664101950015208 0.5749366211965874 -0.7736106834876981 1.0793197697190398
  position -1.1627749036360324 1.4867983451225752 1.279381697559354
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 2 2
}
Robot {
  children [
    DEF Body Pose {
      translation 0 0 0.04
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.25 0.15 0.05
          }
        }
      ]
    }
    DEF front_wheel_1 HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0.075 -0.078 0.025
      }
      device [
        RotationalMotor {
          name "front_motor_1"
        }
      ]
      endPoint Solid {
        translation 0.075 -0.078 0.025
        rotation 1 0 0 1.57
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
              baseColor 1 0 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.01
              radius 0.025
            }
          }
        ]
        name "front_wheel_1"
        boundingObject USE wheel
        physics DEF wheel_physics Physics {
        }
      }
    }
    DEF front_wheel_2 HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0.075 0.078 0.025
      }
      device [
        RotationalMotor {
          name "front_motor_2"
        }
      ]
      endPoint Solid {
        translation 0.075 0.078 0.025
        rotation 1 0 0 1.57
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
              baseColor 1 0 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.01
              radius 0.025
            }
          }
        ]
        name "front_wheel_2"
        boundingObject USE wheel
        physics DEF wheel_physics Physics {
        }
      }
    }
    DEF back_wheel_1 HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor -0.075 -0.078 0.025
      }
      device [
        RotationalMotor {
          name "back_motor_1"
        }
      ]
      endPoint Solid {
        translation -0.075 -0.078 0.025
        rotation 1 0 0 1.57
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
              baseColor 1 0 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.01
              radius 0.025
            }
          }
        ]
        name "back_wheel_1"
        boundingObject USE wheel
        physics DEF wheel_physics Physics {
        }
      }
    }
    DEF back_wheel_2 HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor -0.075 0.078 0.025
      }
      device [
        RotationalMotor {
          name "back_motor_2"
        }
      ]
      endPoint Solid {
        translation -0.075 0.078 0.025
        rotation 1 0 0 1.57
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
              baseColor 1 0 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.01
              radius 0.025
            }
          }
        ]
        name "back_wheel_2"
        boundingObject USE wheel
        physics DEF wheel_physics Physics {
        }
      }
    }
    Solid {
      translation 0.125 0.025 0.045
      children [
        DEF eye_1 Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.007 0.007 0.007
          }
        }
      ]
      name "eye_1"
    }
    Solid {
      translation 0.125 -0.025 0.045
      children [
        DEF eye_2 Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.007 0.007 0.007
          }
        }
      ]
      name "eye_2"
    }
    DistanceSensor {
      translation 0.03 -0.078 0.045
      rotation 0 0 1 -1.57
      children [
        DEF distance_sensor Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "distance_sensor_right_1"
      boundingObject USE distance_sensor
      physics Physics {
      }
      lookupTable [
        0 0 0
        0.2 1000 0
      ]
    }
    DistanceSensor {
      translation 0.03 0.078 0.045
      rotation 0 0 1 1.57
      children [
        DEF distance_sensor Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "distance_sensor_left_1"
      boundingObject USE distance_sensor
      physics Physics {
      }
      lookupTable [
        0 0 0
        0.2 1000 0
      ]
    }
    DistanceSensor {
      translation -0.03 -0.078 0.045
      rotation 0 0 1 -1.57
      children [
        DEF distance_sensor Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "distance_sensor_right_2"
      boundingObject USE distance_sensor
      physics Physics {
      }
      lookupTable [
        0 0 0
        0.2 1000 0
      ]
    }
    DistanceSensor {
      translation -0.03 0.078 0.045
      rotation 0 0 1 1.57
      children [
        DEF distance_sensor Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "distance_sensor_left_2"
      boundingObject USE distance_sensor
      physics Physics {
      }
      lookupTable [
        0 0 0
        0.2 1000 0
      ]
    }
    DistanceSensor {
      translation 0.125 0.025 0.045
      rotation 0 0 1 0.5235996938995747
      children [
        DEF distance_sensor Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "distance_sensor_front_left"
      boundingObject USE distance_sensor
      physics Physics {
      }
      lookupTable [
        0 0 0
        0.2 1000 0
      ]
    }
    DistanceSensor {
      translation 0.125 -0.025 0.045
      rotation 0 0 1 -0.5235996938995747
      children [
        DEF distance_sensor Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "distance_sensor_front_right"
      boundingObject USE distance_sensor
      physics Physics {
      }
      lookupTable [
        0 0 0
        0.2 1000 0
      ]
    }
  ]
  boundingObject USE Body
  physics Physics {
  }
  controller "robot_controller_with_distance_sensor"
}
